src/nvim/os/pty_process_win.c

```diff
@@ -182,6 +195,19 @@ static void pty_process_connect_cb(uv_connect_t *req, int status)
   req->handle = NULL;
 }
 
+static void wait_eof_timer_cb(uv_timer_t *wait_eof_timer)
+  FUNC_ATTR_NONNULL_ALL
+{
+  PtyProcess *ptyproc =
+    (PtyProcess *)((uv_handle_t *)wait_eof_timer->data);
```

Comment:
we don't cast `void *` pointers, instead just:

    PtyProcess *ptyproc = wait_eof_timer->data;